src/App.js

```diff
@@ -23,15 +23,19 @@ class App extends Component {
     sideBarActiveLink: "",
   };
 
+  components = {
+    // key name: component, route, acceptable queries
+    BugList: [BugList, "bug-list", ""],
+    NewBug: [NewBug, "new-bug", ""],
+    EditBug: [EditBug, "bug-list", "/:bug_id"],
+    AddUser: [AddUser, "add-user", ""],
+    EditProject: [EditProject, "edit-project", ""],
+  };
+
```

Comment:
You could use object syntax here rather than arrays, that way you can reference everything by key and avoid the need for comments and the ambiguity later.

```suggestion
BugList: {
                  component: NewBug,
                  route: 'bug-list',
                  accepted_queries: ''
}
```

Ideally 'bug-list' should be 'bugs', and 'new-bug' should be 'bugs/new' etc.

Reply:
Yeah i can change it to object syntax, will look a lot cleaner that way.

Ill have a look at changing the routes as well.

Reply:
all good, ive changed it to object syntax and added the changed routes, took a bit of tinkering but its good to go. I'll push the branch and do the pull request myself as it just contains the changes you've suggested here.